source/common/thread_local/thread_local_impl.cc

```diff
@@ -94,6 +94,9 @@ void InstanceImpl::runOnAllThreads(Event::PostCb cb) {
 void InstanceImpl::runOnAllThreads(Event::PostCb cb, Event::PostCb all_threads_complete_cb) {
   ASSERT(std::this_thread::get_id() == main_thread_id_);
   ASSERT(!shutdown_);
+  // Handle main thread first so that when the last worker thread wins, we could just call the
+  // all_threads_complete_cb method.
+  cb();
```

Comment:
probably just note here that you are trading off parallelism for programming simplicity; it's a reasonable tradeoff but I think it should be stated explicitly that you are making it.

Reply:
Added the note.